InteractionBase/src/commands/CAddNodeToViewByName.cpp

```diff
@@ -40,18 +41,42 @@ bool CAddNodeToViewByName::canInterpret(Visualization::Item*, Visualization::Ite
 	return commandTokens.size() >= 1 && commandTokens.first() == name();
 }
 
-CommandResult* CAddNodeToViewByName::execute(Visualization::Item*, Visualization::Item* target,
-		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>&)
+CommandResult* CAddNodeToViewByName::execute(Visualization::Item* source, Visualization::Item* target,
+		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>& cursor)
 {
 	if (commandTokens.size() < 2)
 		return new CommandResult(new CommandError("Please specify a node to add"));
 
+	auto currentView = target->scene()->currentViewItem();
+	QPoint posToInsert;
+	if (source == currentView && isHorizontal(cursor->region()))
+	{
+		auto below =  findItemBelow(currentView, cursor->region());
```

Comment:
Just realized that this actually doesn't work - we are using DynamicGridFormElement and not GridLayoutFormElement for the ViewItem. Anyway, that class already has such a function (focusedElementIndex), however it does not seem to do what would be required here: If I open a command on a child of it, it always returns the position (-1,-1), and not actually the focused item's index.

I'm calling it using grid->focusedElementIndex(currentViewItem). Any idea why it doesn't work?